jctools-core/src/main/java/org/jctools/queues/atomic/BaseLinkedAtomicQueue.java

```diff
@@ -53,6 +53,51 @@ protected final void spConsumerNode(LinkedQueueAtomicNode<E> node) {
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This method is only safe to call from the consumer thread!
+     */
+    @Override
+    public final boolean remove(Object o) {
+        if (null == o) {
+            throw new NullPointerException();
+        }
+
+        for (LinkedQueueAtomicNode<E> prevConsumerNode = null, currConsumerNode = getNextConsumerNode(lpConsumerNode());
+             currConsumerNode != null;
+             prevConsumerNode = currConsumerNode, currConsumerNode = getNextConsumerNode(currConsumerNode)) {
+            if (o.equals(currConsumerNode.lpValue())) {
+                LinkedQueueAtomicNode<E> nextNode = getNextConsumerNode(currConsumerNode);
+                if (nextNode != null && prevConsumerNode != null) {
+                    // We are removing an interior node.
+                    prevConsumerNode.soNext(nextNode);
+                } else if (prevConsumerNode == null && nextNode == null) {
+                    // We are removing the only node.
+                    currConsumerNode.spValue(null);
+                    spConsumerNode(currConsumerNode);
+                } else if (nextNode == null) {
+                    // We are removing the last node.
```

Comment:
I prefer to use the terms consumer/producer nodes rather than last/first tail/head.